client/signup/steps/site-info-collection/index.tsx

```diff
@@ -0,0 +1,277 @@
+						value: formValues.instagramUrl,
+						name: 'instagramUrl',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+		{
+			title: translate( '%d. Contact Information', {
+				args: [ 4 ],
+				comment: 'This is the serial number: 4',
+			} ),
+			component: (
+				<ContactInformation
+					displayEmailProps={ {
+						value: formValues.displayEmail,
+						name: 'displayEmail',
+					} }
+					displayPhoneProps={ {
+						value: formValues.displayPhone,
+						name: 'displayPhone',
+					} }
+					displayAddressProps={ {
+						value: formValues.displayAddress,
+						name: 'displayAddress',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+	];
+
+	const isSectionAtIndexTouchedInitialState: Record< string, boolean > = {};
+	for ( let i = 0; i <= currentIndex; i++ ) {
+		isSectionAtIndexTouchedInitialState[ `${ i }` ] = true;
+	}
+
+	const [ isSectionAtIndexATouched, setIsSectionAtIndexATouched ] = useState<
+		Record< string, boolean >
+	>( isSectionAtIndexTouchedInitialState );
+
+	const runValidation = ( validator: ValidatorFunction ) => {
+		const validationResult = validator( formValues );
+		setFormErrors( {
+			...formErrors,
+			...validationResult.errors,
+		} );
+		return validationResult;
+	};
+
+	const submitStep = () => {
+		for ( let index = 0; index < sections.length; index++ ) {
+			const section = sections[ index ];
+			if ( section.validate ) {
+				const validationResult = runValidation( section.validate );
+				if ( ! validationResult.result ) {
+					dispatch( updateSiteInfoCurrentIndex( index ) );
+					return;
+				}
+			}
+		}
+
+		const step = {
+			stepName,
+			...additionalStepData,
+		};
+		submitSignupStep( step, {
+			...formValues,
+		} );
+		goToNextStep();
+	};
+
+	const onOpen = ( currentIndex: number ) => {
+		dispatch( updateSiteInfoCurrentIndex( currentIndex ) );
+		dispatch( updateSiteInfoValues( formValues ) );
+		setIsSectionAtIndexATouched( { ...isSectionAtIndexATouched, [ `${ currentIndex }` ]: true } );
+	};
+
+	const onNext = ( validator?: ValidatorFunction ) => {
+		dispatch( updateSiteInfoValues( formValues ) );
+
+		if ( validator ) {
+			const validationResult = runValidation( validator );
+			if ( ! validationResult.result ) {
+				return;
+			}
+		}
+
+		if ( currentIndex < sections.length - 1 ) {
+			dispatch( updateSiteInfoCurrentIndex( currentIndex + 1 ) );
+			setIsSectionAtIndexATouched( {
+				...isSectionAtIndexATouched,
+				[ `${ currentIndex + 1 }` ]: true,
+			} );
+		} else {
+			submitStep();
+		}
+	};
+
+	return (
+		<>
+			{ sections.map( ( section, index ) => (
+				<SectionWrapper
+					key={ `${ index }` }
+					title={ section.title }
+					summary={ section.summary }
+					isExpanded={ index === currentIndex }
+					showSkip={ section.showSkip }
+					component={ section.component }
+					isTouched={ isSectionAtIndexATouched[ `${ index }` ] }
```

Comment:
A small addendum to a comment above, I believe the touched condition here can be derived with
`index < currentIndex`

Reply:
Ok this comment is kind of obsolete from your  comment above :D